### PR TITLE
feat(api): expose credentials in GET /integrations/:uniqueKey

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Changes

- `GET /integrations/:uniqueKey` now expose credentials (optional)
Some customers need the credentials for internal operations or oauth revoke. This expose optionally the credentials with some added security.

- Try to fix IntegrationConfig type by creating a smaller scope for integration create
This type is annoying, it's used a bit everywhere and everything is marked as undefined except it's null not undefined. Anyway, first steps since I needed better types.

### CODE REVIEW

**Review by nalanj — APPROVED** [review]
Looks good. Tested it locally and it worked great too.

File: packages/types/lib/utils.ts [hunk 1]
```diff
@@ -12,3 +12,9 @@ export interface Logger {
 
 type ValidateSelection<T, U> = U extends T ? U : never;
 export type PickFromUnion<T, U extends T> = ValidateSelection<T, U>;
+
+export type NullablePartial<
+    TBase,
+    NK extends keyof TBase = { [K in keyof TBase]: null extends TBase[K] ? K : never }[keyof TBase],
```
Comment (TBonnin) [inline packages/types/lib/utils.ts]: I would rename `NK` `TNullableKey` to follow the same pattern and make it a bit more descriptive

File: packages/types/lib/utils.ts [hunk 2]
```diff
@@ -12,3 +12,9 @@ export interface Logger {
...
+    NP = Partial<Pick<TBase, NK>> & Pick<TBase, Exclude<keyof TBase, NK>>
+> = { [K in keyof NP]: NP[K] };
```
Comment (TBonnin) [inline packages/types/lib/utils.ts]: not sure you need NP. can we have 
```
export type NullablePartial<
   TBase, 
   TNullableKeys ...
> = Omit<TBase, TNullableKeys> & Partial<Pick<TBase, TNullableKeys>>
```

**Review by TBonnin — APPROVED** [review]
code looks good but I am not sure I understand the purpose of exposing integration credentials to the users when they are the ones who are providing those credentials in the first place

File: packages/types/lib/utils.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/types/lib/utils.ts]: You forgot to add the non nullable keys, but yes I can simplify indeed

**Pull request merged**